src/KafkaFlow/Configuration/ConsumerConfigurationBuilder.cs

```diff
@@ -234,19 +234,22 @@ public IConsumerConfiguration Build(ClusterConfiguration clusterConfiguration)
             var middlewareConfiguration = this.middlewareConfigurationBuilder.Build();
 
             this.consumerConfig ??= new ConsumerConfig();
-            this.consumerConfig.BootstrapServers ??= string.Join(",", clusterConfiguration.Brokers);
-            this.consumerConfig.GroupId ??= this.groupId;
-            this.consumerConfig.AutoOffsetReset ??= this.autoOffsetReset;
-            this.consumerConfig.MaxPollIntervalMs ??= this.maxPollIntervalMs;
-            this.consumerConfig.StatisticsIntervalMs ??= this.statisticsInterval;
 
-            this.consumerConfig.EnableAutoOffsetStore = false;
-            this.consumerConfig.EnableAutoCommit = false;
+            var consumerConfigCopy = new KafkaFlowConsumerConfig(this.consumerConfig);
```

Comment:
Why do we need this config copy?

Reply:
When the user passes the same ConsumerConfig to multiple consumers, the consumerConfig reference is shared between all the consumer builders. When the framework is setting the ConsumerConfiguration, it will load the settings passed through the builder methods (ex.: WithGroupId). As the ConsumerConfig shares the same reference between all consumers, the settings will be shared between all consumers instead of being independent.